pkg/envoy/lds.go

```diff
@@ -101,3 +146,261 @@ func NewInboundListenerCheck(configGetter ConfigGetter, osmVersion osm.Controlle
 		expectedListenersPerVersion: osm.InboundListenerNames,
 	}
 }
+
+// Run implements common.Runnable
+func (l ListenerFilterCheck) Run() outcomes.Outcome {
+	// Check if permissive mode is enabled, in which case every meshed pod is allowed to communicate with each other
+	if l.cfg.IsPermissiveTrafficPolicyMode() {
+		return outcomes.Info{Diagnostics: "OSM is in permissive traffic policy modes -- all meshed pods can communicate and SMI access policies are not applicable"}
+	}
+
+	// Get rule types from TrafficTarget. The rules will be used to determine what filter chains are expected in the src and dst Envoy configs
+	var ruleTypes map[string]struct{}
+	var err error
+	switch l.osmVersion {
+	case "v0.5", "v0.6":
+		ruleTypes, err = getRuleTypesFromMatchingTrafficTargetsV1alpha2(l.srcPod, l.dstPod, l.accessClient)
+	case "v0.7", "v0.8", "v0.9":
+		ruleTypes, err = getRuleTypesFromMatchingTrafficTargetsV1alpha3(l.srcPod, l.dstPod, l.accessClient)
+	default:
+		return outcomes.Fail{Error: fmt.Errorf(
+			"OSM Controller version could not be mapped to a TrafficTarget version. Supported versions are v0.5 through v0.9")}
+	}
+	if err != nil {
+		return outcomes.Info{Diagnostics: fmt.Sprintf(
+			"Pod '%s/%s' is not allowed to communicate to pod '%s/%s' via any SMI TrafficTarget policy.\n",
+			l.srcPod.Namespace,
+			l.srcPod.Name,
+			l.dstPod.Namespace,
+			l.dstPod.Name)}
+	}
+	if len(ruleTypes) == 0 {
+		return outcomes.Info{Diagnostics: fmt.Sprintf(
+			"No applicable Traffic Targets in namespace %s to check routes for, or no rules specified in Traffic Targets",
+			l.dstPod.Namespace)}
+	}
```

Comment:
Should these be `outcomes.Fail` instead? The user would be running `connectivity pod-to-pod pod1 pod2`, so they'd expect failures.

Reply:
In the TrafficTargetCheck these cases are treated as outcomes.Info. In other similar cases if no traffic target is found the outcome is info. I think this should be a failed outcome if we want the connectivity check to fail if the pods are not configured correctly to connect and/or if the pods don't have a shared traffic target. Right now we don't consider the latter a failure.